docs/adr/0007-scorecard-metrics.md

```diff
@@ -0,0 +1,25 @@
+# Aggregating OpenSSF Scorecard metrics to compute software stack quality scores
+
+* Status: proposed
+* Date: 2022-August-1
+
+Technical Story: Aggregate OpenSSF Scorecard metrics on a package release update to compute software stack quality scores.
+
+## Context and Problem Statement
+
+As part of advise output improvements mentioned in [](https://github.com/thoth-station/core/issues/434), we would like to aggregate Security Scorecards data on a new package release to provide software stack quality metrics with regards to the global quality of packages that can be found in Thoth's Database. These metrics would inform the user about the health of a project dependencies and provide a software stack score (on a 0-100 scale or A,B,C...) based on Scorecard data.
+
+**Note:** The architecture diagram below supposes that the OSSF Scorecards dataset available on BigQuery will expose metrics by package release instead of repository head commit SHA as it is currently the case. However if this feature is not going to be implemented on the Scorecards project side, we will need to handle the head commit SHA to release association logic by ourselves and thus revisit the diagram to include appropriate calls to the GitHub API and additional computations.
+
+The data aggregation and score computation logic would be implemented as follows:
+
+* With each `package-releases-job` run, aggregate data about the latest package release and corresponding Scorecards data retrieved from BigQuery
```

Comment:
As we want to aggregate data about packages we do not have yet into the prescriptions database, I don't think this will work unless we modify the scorecards handler logic. Another alternative to this architecture would be to compute prescriptions from new releases aggregated on `package-release-job` without storing this data in the DB, which will (if I'm not mistaken) greatly increase the number of prescriptions we already have. The global quality metrics could then be derived from prescriptions directly and stored in the DB on a regular job run for example.